lib/hypertrack.dart

```diff
@@ -1,23 +1,29 @@
 import 'package:flutter/services.dart';
-import 'package:hypertrack_plugin/hypertrack_platform_interface.dart';
+import 'package:hypertrack_plugin/data_types/error.dart';
+import 'package:hypertrack_plugin/data_types/result.dart';
+import 'package:hypertrack_plugin/src/sdk_method.dart';
+import 'package:hypertrack_plugin/src/serialization/availability.dart';
+import 'package:hypertrack_plugin/src/serialization/error.dart';
+import 'package:hypertrack_plugin/src/serialization/tracking_state.dart';
 
-import 'const/constants.dart';
+import 'data_types/location.dart';
 
-/// Plugin allows you to use your application as a location data source
-/// feeding HyperTrack platform.
+/// This plugin allows you to use Hypertrack SDK for Flutter apps to get realtime device location
 class HyperTrack {
-  final HypertrackPlatformInterface _pluginInterface =
-      HypertrackPlatformInterface.instance;
+  HyperTrack._();
 
   /// Use this method to get the SDK instance.
   ///
-  /// SDK will connect to the account identified by [publishableKey].
-  Future<HyperTrack> initialize(String publishableKey) =>
-      _pluginInterface.initialize(publishableKey);
+  /// SDK will use the account identified by [publishableKey].
+  static Future<HyperTrack> initialize(String publishableKey) {
+    return invokeSdkVoidMethod(SdkMethod.initialize, publishableKey)
+        .then((value) => HyperTrack._());
+  }
 
   /// Returns string that uniquely identifies device in HyperTrack platform.
-  Future<String> getDeviceId() async =>
-      HypertrackPlatformInterface.instance.getDeviceId();
+  Future<String> get deviceId async {
```

Comment:
```suggestion
  Future<String> get deviceID async {
```

Reply:
https://dart.dev/guides/language/effective-dart/style#do-capitalize-acronyms-and-abbreviations-longer-than-two-letters-like-words